File: Week03/Code/Vectorize1.R
M<-matrix(runif(1e6),1e3,1e3)

SumAllElements<-function(M){
  Dimensions<-dim(M)
  Tot<-0
  for(i in 1:Dimensions[1]){
    for(j in 1:Dimensions[2]){
      Tot<-Tot+M[i,j]
    }
  }
  return(Tot)
}

cat("Using loops, the time taken is:\n")
cat(paste0(round(unname(system.time(SumAllElements(M))[1]),3),"\n"))

cat("Using the built vectorized function, the time taken is:\n")
cat(paste0(round(unname(system.time(sum(M))[1]),3),"\n"))

File: Week03/Code/Vectorize1.py
#!/bin/env python3

## imports
import random
import time
import numpy

## variables
M=[[random.random() for x in range(1000)]for y in range(1000)]

## func
def SumAllElements(M):
    Dimensions=[len(M),len(M[0])]
    Tot=0
    for i in range(Dimensions[0]):
        for j in range(Dimensions[1]):
            Tot=Tot+M[i][j]
    return(Tot)

print("Using loops, the time taken is:")
start=time.time()
SumAllElements(M)
print("%s" %(round(time.time()-start,3)))

print("Using the built vectorized function, the time taken is:")
start=time.time()
numpy.sum(M)
print("%s" %(round(time.time()-start,3)))